src/mesh/Mesh.hpp

```diff
@@ -218,7 +222,26 @@ class Mesh {
     return _communicationMap;
   }
 
-  void addMesh(Mesh &deltaMesh);
+  template <typename UnaryPredicate>
+  void filterAndAddMesh(Mesh const &other, UnaryPredicate filter)
+  {
+    PRECICE_TRACE();
+    PRECICE_ASSERT(_dimensions == other.getDimensions());
+
+    FilteredVertices vertexMap = filterAndAddVertices(other, filter);
+    FilteredEdges    edgeMap   = filterAndAddEdges(other, vertexMap);
```

Comment:
We can declare them as immutable now.

```suggestion
    const FilteredVertices vertexMap = filterAndAddVertices(other, filter);
    const FilteredEdges    edgeMap   = filterAndAddEdges(other, vertexMap);
```